ui/src/views/Provisioners/PendingTasks/index.jsx

```diff
@@ -0,0 +1,160 @@
+import TableCellItem from '../../../components/TableCellItem';
+import DateDistance from '../../../components/DateDistance';
+import Label from '../../../components/Label';
+import Breadcrumbs from '../../../components/Breadcrumbs';
+
+@graphql(pendingTasks, {
+  options: props => ({
+    variables: {
+      taskQueueId: `${props.match.params.provisionerId}/${props.match.params.workerType}`,
+      connection: {
+        limit: VIEW_WORKER_POOL_PENDING_TASKS_PAGE_SIZE,
+      },
+    },
+  }),
+})
+export default class WMViewPendingTasks extends Component {
+  handlePageChange = ({ cursor, previousCursor }) => {
+    const {
+      data: { fetchMore },
+      match: { params },
+    } = this.props;
+
+    return fetchMore({
+      query: pendingTasks,
+      variables: {
+        taskQueueId: `${params.provisionerId}/${params.workerType}`,
+        connection: {
+          limit: VIEW_WORKER_POOL_PENDING_TASKS_PAGE_SIZE,
+          cursor,
+          previousCursor,
+        },
+      },
+      updateQuery(previousResult, { fetchMoreResult: { listPendingTasks } }) {
+        // use dotProp.set to avoid lint warning about assigning to properties
+        return dotProp.set(
+          previousResult,
+          'listPendingTasks',
+          listPendingTasks
+        );
+      },
+    });
+  };
+
+  get workersLink() {
+    const { provisionerId, workerType } = this.props.match.params;
+
+    return `/provisioners/${encodeURIComponent(
+      provisionerId
+    )}/worker-types/${encodeURIComponent(workerType)}`;
+  }
+
+  renderRow({ node: { taskId, runId, inserted, task } }) {
+    return (
+      <TableRow key={taskId}>
+        <TableCell>
+          <Link to={`/tasks/${taskId}`}>
+            <TableCellItem button>
+              {taskId}
+              <LinkIcon size={16} />
+            </TableCellItem>
+          </Link>
+        </TableCell>
+        <TableCell>{runId}</TableCell>
+        <TableCell>
+          <Label mini status="info">
+            {task.priority}
+          </Label>
+        </TableCell>
+        <TableCell>
+          <DateDistance from={new Date(inserted)} />
+        </TableCell>
+        <TableCell>{task.metadata?.name}</TableCell>
+      </TableRow>
+    );
+  }
+
+  render() {
+    const {
+      data: { loading, error, listPendingTasks },
+    } = this.props;
+    const { provisionerId, workerType } = this.props.match.params;
+
+    return (
+      <Dashboard
+        title={`Pending tasks in "${provisionerId}/${workerType}"`}
+        disableTitleFormatting>
+        <Box
+          sx={{
+            display: 'flex',
+            flexWrap: 'wrap',
+            alignItems: 'center',
+            width: '100%',
+          }}>
+          <div style={{ flexGrow: 1, marginRight: 8 }}>
+            <Breadcrumbs>
+              <Link to="/provisioners">
+                <Typography variant="body2">Workers</Typography>
+              </Link>
+              <Link to={`/provisioners/${provisionerId}`}>
```

Comment:
Same question in this file about URI encoding.

Reply:
Yeah, I guess for the sake of consistency we should  either encode everywhere or nowhere.. Valid comment.
`provisionerId`, `workerGroup`, `workerType`, `workerId` have very strict validation in schema:

```
identifier-pattern:     ^([a-zA-Z0-9-_]*)$
provisionerid-pattern: '^[a-zA-Z0-9-_]{1,38}$'
workertype-pattern: '^[a-z]([-a-z0-9]{0,36}[a-z0-9])?$'
```

which makes it easier for us to ignore urlencoding as they wouldn't contain "unfriendly" characters.. So I guess I can drop that encoding from other places here for consistency